filebeat/module/kibana/log/ingest/pipeline-ecs.yml

```diff
@@ -0,0 +1,21 @@
+description: Pipeline for parsing Kibana ecs logs
+processors:
+- set:
+    field: event.ingested
+    value: '{{_ingest.timestamp}}'
+- set:
+    copy_from: '@timestamp'
+    field: event.created
+- script:
+    lang: painless
+    inline: 'ctx.json.keySet().each (key -> ctx[key] = ctx.json.get(key))'
```

Comment:
I was surprised to see this rather than https://www.elastic.co/guide/en/elasticsearch/reference/master/json-processor.html - wondering what's special about the kibana module that causes them to land like this rather than as a `message` JSON string like they land in the elasticsearch module. 

Guess I'll give it a go and see if I can work out what's happening.

Reply:
Ah I see. It's because of https://github.com/elastic/beats/blob/cca59bad174b56ced2f74bccc6694f233752eecd/filebeat/module/kibana/log/config/log.yml#L8

I tried doing something like this but the fields end up as just strings rather than actual values.

```
- foreach:
    field: json
    processor:
      set:
        field: _ingest._key
        value: "{{_ingest._value}}"
```

Guessing this is the best approach. Would be nice to have an ingest pipeline SME weigh-in. Maybe @jbaiera can recommend one?

Reply:
Sorry for late response! Yeah, I don't think there is an easier way to merge two object fields together in ingest at the moment. You could use the append processor, but I think you'd need to know all the field names ahead of time.

Reply:
No worries @jbaiera ! thanks for the cross check.

Maybe interesting food for thought, but the lack of such a thing makes me wonder if we should stick to json parsing at the ES side (since the https://www.elastic.co/guide/en/elasticsearch/reference/current/json-processor.html can merge/expand objects).